minigunapp/views.py

```diff
@@ -0,0 +1,62 @@
+import json
+import yaml
+
+from django.http import HttpResponse, Http404, JsonResponse
+from django.shortcuts import render
+from cerberus import Validator
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import authentication, permissions
+
+from minigunapp.models import Email
+from minigunapp.serializers import EmailSerializer
+from minigunapp.tasks import send_email
+
+
+v = Validator()
+schema = yaml.load(open('minigunapp/schemas/email.yaml'))
+page_size = 10
+
+
+class EmailList(APIView):
+
+    def get(self, request, format=None):
+        """ Returns a single page of emails in Json format """
+
+        page = int(request.GET.get('page', 1))
+        page = page if page > 0 else 1
+        start = page_size * page
+        end = page_size * (page + 1)
+
+        emails = Email.objects[start:end]
+        emails = [EmailSerializer(email).data for email in emails]
+        return JsonResponse(emails, safe=False)
+
+    def post(self, request, format=None):
+        """ Creates and saves new email, then dispatches send task to celery.
+        Returns the new email object.
+        """
+
+        body = json.loads(request.body.decode("utf-8"))
+        valid = v.validate(body, schema)
+        if(not valid):
+            return JsonResponse(v.errors, status=400)
+
+        email = Email(**body)
+        email.from_ = 'nclow@localhost.localdomain'
+        email.status = 'pending'
+        email.save()
+
```

Comment:
If it were killed before line 48, the email object would never get saved to the database, and the system would lose that email. If it were killed between 48 and 49, the system would keep the data, but no task would ever be dispatched. So yes, the Email object in the database would be stuck in 'pending'.